lib/productions/iterable.js

```diff
@@ -0,0 +1,48 @@
+import { Base } from "./base";
+import { type_with_extended_attributes } from "./helpers";
+
+export class IterableLike extends Base {
+  /**
+   * @param {import("../tokeniser.js").Tokeniser} tokeniser
+   */
+  static parse(tokeniser) {
+    const start_position = tokeniser.position;
+    const tokens = {};
+    const ret = new IterableLike({ source: tokeniser.source, tokens });
+    tokens.readonly = tokeniser.consume("readonly");
+    tokens.base = tokens.readonly ?
+      tokeniser.consume("maplike", "setlike") :
+      tokeniser.consume("iterable", "maplike", "setlike");
+    if (!tokens.base) {
+      tokeniser.unconsume(start_position);
+      return;
+    }
+
+    const { type } = ret;
+    const secondTypeRequired = type === "maplike";
+    const secondTypeAllowed = secondTypeRequired || type === "iterable";
+
+    tokens.open = tokeniser.consume("<") || tokeniser.error(`Missing brackets in ${type} declaration`);
```

Comment:
“Missing less-than sign (<)”, the word “bracket” is too ambiguous. Can do the same below for the closing greater than.